### DISCUSSION TITLE
ESP32 3.0.0 RC1 with ETH for W5500

### DISCUSSION THREAD

**Discussion by kapyaar** [Bodmer/TFT_eSPI#3298] [General]
Hi,

I am working with the 3.0.0 RC1 for esp32 Arduino. I have a W5500 board, and ST7735 1.8" tft display. I have them all working on their own. But when I put them all together, it looks like there is an issue with the SPI instance. The ETH part seems to use "SPI" while the tft uses "spi".

This is the ETH code I am using.
https://github.com/espressif/arduino-esp32/blob/master/libraries/Ethernet/examples/ETH_W5500_Arduino_SPI/ETH_W5500_Arduino_SPI.ino

When I add the tft part to this example, If I am not mistaken, the spi port definition is happening in TFT_eSPI_ESP32.c for my setup, and the following line seems to get selected.
SPIClass spi = SPIClass(FSPI);

Whereas in ETH.h, they have defined the port as SPI. I saw https://github.com/Bodmer/TFT_eSPI/discussions/3129#discussioncomment-8232166 but wanted to check if this was something that may have a quick solution that I could use?

 I noticed issue #1991 but I was not sure How I should go about in this particular case. Any guidance much appreciated.

**Comment by infrafast** [comment]
Hello same issue for me, if you have any solution to this, would be great if you could share. Thx

**Comment by kapyaar** [comment]
For my use case, it was fixed with the updated ETH library for ESP32. Currently, I use their release version 3.1.3.

**Reply by infrafast** [reply]
thanks, in ly case I use Ethernet.h , so if I add #define USE_HSPI_PORT it solve the TFT issue but the ethernet w5500 stops working. I think I have to also instruct the driver to use HSPI as well and not FSPI
